text/0049-network-ids.md

```diff
@@ -0,0 +1,191 @@
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This adds more configuration to the network.
+
+Clients may send a few more bytes with each request.
+
+This proposal does NOT, at this time, propose to hash the network id into the responder id. While that would logical, it is also an enclave change.
+
+This means that if there is a configuration problem, it's possible that not all nodes in the network have the same network id, but in this case,
+we can fall-back to the attestation trust roots to prevent a problem, and no harm was done relative to status quo. Eventually users who connect
+to the misconfigured servers will be unable to make progress and report an error, and the configuration can be fixed.
+
+This does not, at this time, propose to make the same API change in the Fog servers. Putting this in the Fog servers would mean that balance checks
+cannot succeed without getting this error if there is mixing of test and prod, so Fog clients would fail faster in case of mixing networks.
+But just putting it in consensus seems higher priority because that prevents loss of funds, and minting problems.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The primary alternative is, do nothing:
+
+* Continue to use ad-hoc configuration differences like attestation trust roots and token ids as the primary means to distinguish dev, test, prod environments.
+* Continue to introduce more such ad-hoc configuration differences as more features are added.
+* Cross-chain bridges primarily lean on attestation roots to avoid e.g. accidentally minting on the wrong network, but it is a bit harder to manage the configuration,
+  because attestation roots live in files.
+
+Another alternative: The Ethereum [JSON-RPC API](https://ethereum.org/en/developers/docs/apis/json-rpc/) returns the network id as a string, but in practice all the ids are actually integers.
+The mapping from network names to ids is available at chainlist.org.
+
+* Integers may be a bit smaller on the wire.
+* Assigning consecutive numbers instead of names creates additional work around an additional service and is somewhat unfriendly to ad-hoc dev networks and local networks.
+* Size on the wire is likely not a serious concern. But if it were, another approach would be to send a hash of the network id string, so that we can ensure that the on-the-wire size is bounded,
+  and the error message in case of a mismatch sends back the full string. OTOH it seems that in real life the hash would probably not be shorter than the string network id.
+
+Ethereum actually has two ids like this: A network id, and a chain id.
+
+* Peer-to-peer communication between nodes uses the network id
+* Transaction signatures use the chain id
+
+For most networks the two values are the same.
+Chain ID was introduced in EIP-155 when the Ethereum Classic fork occurred, to prevent mixing of these two networks.
```

Comment:
```suggestion
Chain ID was introduced in [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md) when the Ethereum Classic fork occurred, to prevent mixing of these two networks.
```